DXMainClient/Domain/Multiplayer/MapLoader.cs

```diff
@@ -205,8 +211,16 @@ private void CacheCustomMaps(ConcurrentDictionary<string, Map> customMapCache)
             try
             {
                 var jsonData = File.ReadAllText(CUSTOM_MAPS_CACHE);
-            
-                return JsonConvert.DeserializeObject<ConcurrentDictionary<string, Map>>(jsonData) ?? new ConcurrentDictionary<string, Map>();
+
+                var customMapCache = JsonConvert.DeserializeObject<CustomMapCache>(jsonData);
+
+                var customMaps = customMapCache?.Version == CurrentCustomMapCacheVersion && customMapCache.Maps != null
+                    ? customMapCache.Maps : new ConcurrentDictionary<string, Map>();
```

Comment:
This should allow us to "force clear" someone's cache on an update, should there be an issue in the future or new data to be added to the cache.